server/src/test/java/org/elasticsearch/index/mapper/MappingParserTests.java

```diff
@@ -54,6 +55,20 @@ private static MappingParser createMappingParser(Settings settings, Version vers
             indexSettings,
             indexSettings.getMode().idFieldMapperWithoutFieldData()
         );
+        Supplier<MappingParserContext> mappingParserContextSupplier = () -> new MappingParserContext(
```

Comment:
I think you can remove the `mappingParserContext` above and replace it with `mappingParserContextSupplier.get()`?